src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((groupsMap: GroupsMap<T>, item: T) => {
+    const groupField = String(item[key]);
+    const groupObjectCopy: GroupsMap<T> = { ...groupsMap };
+
+    if (!{}.hasOwnProperty.call(groupObjectCopy, groupField)) {
+      groupObjectCopy[groupField] = [];
+    }
```

Comment:
You can also use the [Nullish coalescing assignment (??=)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment) here:
```suggestion
    groupObjectCopy[groupField] ??= [];
```